sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -285,6 +285,17 @@ private void finalizeTransaction(final @NotNull IScope scope, final @NotNull Hin
 
     event = processReplayEvent(event, hint, options.getEventProcessors());
 
+    if (event != null) {
+      event = executeBeforeSendReplay(event, hint);
+
+      if (event == null) {
+        options.getLogger().log(SentryLevel.DEBUG, "Event was dropped by beforeSendReplay");
+        options
+            .getClientReportRecorder()
+            .recordLostEvent(DiscardReason.BEFORE_SEND, DataCategory.Replay);
+      }
+    }
+
     if (event == null) {
       options.getLogger().log(SentryLevel.DEBUG, "Replay was dropped by Event processors.");
```

Comment:
should we add a client report here, with a EventProcessor reason?

Reply:
we already do that in [processReplayEvent](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryClient.java#L530)